### PR TITLE
fix(xmr): fix tx sending to an integrated address

### PR DESCRIPTION
**Pull request by ph4r05**
Fixes sending tx to an integrated address as described in #2213

@matejcik

### CODE REVIEW

File: crypto/monero/base58.c [hunk 1]
```diff
@@ -48,7 +48,7 @@ const size_t encoded_block_sizes[] = {
 const size_t full_block_size =
     sizeof(encoded_block_sizes) / sizeof(encoded_block_sizes[0]) - 1;
 const size_t addr_checksum_size = 4;
-const size_t max_bin_data_size = 64;
+const size_t max_bin_data_size = 72;
```
Comment (ph4r05) [inline crypto/monero/base58.c:51]: fuzzing re-run may be needed

**Comment by ph4r05** [comment]
@matejcik I've added a new testing binary https://github.com/ph4r05/monero/releases/tag/v0.18.1.0-dev-tests-u18.04-01

- uses current master
- fixes test using integrated address
- correctly builds transaction with view tags

**Comment by ph4r05** [comment]
hello @matejcik, hope everything is going fine.

I was wondering, is this PR OK? Or would you prefer me merging it to the #2478?

**Review by matejcik — APPROVED** [review]
utACK from me.

@onvej-sl please review the change too, it looks to me that the change is OK but I'd appreciate another set of eyes.

**Comment by ph4r05** [comment]
thanks @matejcik ! @onvej-sl when this PR is merged I will start removing old monero code from pre-BP+ era - will save few kBs from the firmware

**Comment by ph4r05** [comment]
@onvej-sl ping, pls? :)

**Review by mmilata — APPROVED** [review]
LGTM, just modifies the size of a buffer that the rest of the code don't make any assumptions about. Ran the fuzzer for couple M iterations and did not find any problem. Sorry for the wait:(

**Comment by ph4r05** [comment]
Thanks @mmilata ! 🙏  I will try to remove old XMR code (mainly BP, old BP constants) during the week.

**Comment by onvej-sl** [comment]
@ph4r05 I apologize for being unresponsive, I must have missed the github notification.

Anyway, as I told @matejcik earlier today, I don't usually have the capacity to review the monero code anymore.

**Comment by grdddj** [comment]
> when this PR is merged I will start removing old monero code from pre-BP+ era - will save few kBs from the firmware

Will you be looking at old/unused code in `micropython` or in `C`? I have scanned the `micropython` code for unused functions with [this](https://github.com/trezor/trezor-firmware/issues/2515) result (quite a lot of `Monero` functions mostly from `bulletproof.py`), and it is already part of https://github.com/trezor/trezor-firmware/pull/2525

**Pull request merged**